mycroft/skills/intent_service.py

```diff
@@ -332,12 +373,23 @@ def _converse(self, utterances, lang, message):
                               if time.time() - skill[
                                   1] <= self.converse_timeout * 60]
 
+
+        # first check for system levl skills
+        tmp_active_skills = copy(self.active_skills)
+        for skill in tmp_active_skills:
+            if skill[2] == 'system':
```

Comment:
what are skill[2] and skill[0].  improve readability by adding a statement that names the values in these indices.

Reply:
or using NamedTuple instead of a regular tuple for the active_skills

Reply:
This also brings up a really good issue regarding how the PR process is handled and I am interested in input from both of you regarding this concept. 

BTW I agree a named tuple here would be most appropriate and I also agree additional documentation would be beneficial, however, you do realize this code pre-dates me. 

The use of "skill[some magic number] " is something which was here before I touched this code. I simply added an additional value to this list entry but now it seems like I own it and should go back through the code and clean up this old code because I am submitting a PR against it. 

Where does that stop. Suppose I do create a named tuple and refactor any references to skill[] to use that. Now in so doing I may touch another file which also has some questionable code in it. Should I also be required to correct that as a part of this PR process? What if that leads me to yet another file and so on. Also of concern is the fact that I work here. I get paid by the hour so I can take the time to respond to every comment against my PRs, I can address other people's opinions about how I decided to architect something, but contributors do not have that luxury. For most members of our community this is not their full time job and they don't get paid for this effort.

In other words, regarding PRs and our code review process, what is the objective here and what are the trade-offs? Do we want programmers to refactor the code with each PR at the expense of time or do we want to get more code in at the expense of leaving old ugly code ugly. 

I would be interested in both your thoughts.

Reply:
I'm sure I would have reacted to this other than to point out that NamedTuples are a nice alternative to variables for 0 and 1 :) (I'm a bit in love with them) since they work just as normal tuples so no need to change everywhere at once.

To answer your question, I try to keep PR's focused to as small an area as possible, it makes. If I get the opportunity to clean up something related to my update (notice a bad docstring, log message, small structure that can be re-ordered) , I do it in a separate commit. Never cross the beams! (refactoring & feature)

In short, you're not required to refactor / take ownership on any code you basically only move around or extend.

Reply:
What's interesting is I think Jarbas alerted me to the issue that changing this to a named tuple would be a breaking change.

Reply:
Ok, I can't see it but maybe I'm wrong, it wouldn't be the first time and won't be the last time.

Reply:
Existing old code refers to these values using the skill[offset] style but if I change them to a named tuple won't the old code have to be changed to refer to the entries at offset by name? Can you still refer to the individual elements of a named tuple by offset (int) and pass the linter? I could be wrong here.

Reply:
I'm going to leave this alone for now but I agree with the sentiment. Note the structure is documented in the existing comment around line 124 where the active_skills array is initialized. I added the category to this definition for clarity.

Reply:
Yes named tuples are great that way. both indexes and names work:
```python
>>> MyTup = namedtuple('MyTup', ('alpha', 'beta', 'gamma'))
>>> t = MyTup(1, 2, 3)
>>> t[1]
2
>>> t.beta
2
>>>
```

Leaving alone sounds good.